Make pular python client dependencies optional components (#9719)
apache/pulsar: site2/docs/client-libraries-python.md

$ pip install pulsar-client=={{pulsar:version_number}}
```

> ### Optioanl dependencies

Reviewer: ```suggestion
### Optional dependencies
```

Author: fixed

Author: fixed

> To support aspects like pulsar functions and avro serialization additional optiopnal components can be installed alongside the  `pulsar-client` library

Reviewer: ```suggestion
To support aspects like Pulsar functions or Avro serialization, additional optional components can be installed alongside the  `pulsar-client` library
```

Author: fixed

Author: fixed

```shell
# avro serialization
$ pip install pulsar-client=='{{pulsar:version_number}}[avro]'

> # functions runtime
> $ pip install pulsar-client=='{{pulsar:version_number}}[functions]'
> # all optional components

Reviewer: ```suggestion

# functions runtime
$ pip install pulsar-client=='{{pulsar:version_number}}[functions]'

# all optional components
```

Author: fixed

Author: fixed
